src/hooks/useFetch.ts

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useReducer, useRef } from 'preact/hooks';
+
+// Taken from: https://usehooks-ts.com/react-hook/use-fetch
+
+interface State<T> {
+    data?: T;
+    error?: Error;
+}
+type Cache<T> = { [url: string]: T };
+type Action<T> = { type: 'loading' } | { type: 'fetched'; payload: T } | { type: 'error'; payload: Error };
+
+export const useFetch = <T = unknown>(url?: string, fetch_options?: RequestInit): State<T> => {
+    const cache = useRef<Cache<T>>({});
+
+    // Used to prevent state update if the component is unmounted.
+    const cancelRequest = useRef<boolean>(false);
+
+    const initialState: State<T> = { error: undefined, data: undefined };
+
+    const fetchReducer = (state: State<T>, action: Action<T>): State<T> => {
+        switch (action.type) {
+            case 'loading':
+                return { ...initialState };
+            case 'fetched':
+                return { ...initialState, data: action.payload };
+            case 'error':
+                return { ...initialState, error: action.payload };
+            default:
+                return state;
+        }
+    };
+
+    const [state, dispatch] = useReducer(fetchReducer, initialState);
+
+    useEffect(() => {
+        if (!url) return;
+
+        cancelRequest.current = false;
+
+        const fetchData = async () => {
+            dispatch({ type: 'loading' });
+
+            if (cache.current[url]) {
```

Comment:
Maybe we should document somewhere that this fetch can not be used to fetch rapidly changing content, as all subsequent requests will be served from the cache…